samples/text_panes/components/text_pane.js

```diff
@@ -0,0 +1,97 @@
+AFRAME.registerComponent("text_pane", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    // Create textLabel
+    this.textLabel = document.createElement("a-text");
+    this.textLabel.setAttribute("id", "text");
+    this.textLabel.setAttribute("value", data.text[data.currPosition]);
+    this.textLabel.setAttribute("negate", "true");
+    this.textLabel.setAttribute("scale", "2 2 1");
+    this.textLabel.setAttribute("position", "-4.9 0 0.25");
+    this.el.appendChild(this.textLabel);
+
+    if (data.text.length > 1) {
+      const leftNavProp =
+        '{"width": "1.5", "height": "1.5", "depth": "0.01", "color": "white", "x": "-8", "y": "0", "z": "0.005"}';
+      const rightNavProp =
+        '{"width": "1.5", "height": "1.5", "depth": "0.01", "color": "white", "x": "8", "y": "0", "z": "0.005"}';
+
+      // Create left nav textLabel
```

Comment:
For the sake of optimization, let's combine the nav textlabel, and the nav button into one entity. I think aframe has that feature which may be useful. Like:
```
<a-entity
  geometry="primitive: plane; width: 4; height: auto"
  material="color: blue"
  text="value: This text will be 4 units wide."></a-entity>
```
Basically something like the above, but in JS. This means that your `prism` component can be renamed to like `text-box`, and additionally take in some more fields related to the text and its properties and set it appropriately. 
So, instead of needing to keep the box and text in sync, we create an entity that combines the two of them.

Reply:
Combined the textlabel and button into one text-box entity, lmk if this is what you had in mind